engines/tsage/metaengine.cpp

```diff
@@ -154,10 +159,186 @@ class TSageMetaEngine : public AdvancedMetaEngine<TsAGE::tSageGameDescription> {
 		return SaveStateDescriptor();
 	}
 
+	Common::KeymapArray initKeymaps(const char *target) const override;
 };
 
+Common::KeymapArray TSageMetaEngine::initKeymaps(const char *target) const {
+	using namespace Common;
+	using namespace TsAGE;
+
+	Keymap *engineKeyMap = new Keymap(Keymap::kKeymapTypeGame, "tsage-default", _("Default keymappings"));
+	Keymap *gameKeyMap = new Keymap(Keymap::kKeymapTypeGame, "game-shortcuts", _("Game keymappings"));
+	Keymap *blueForceKeyMap = new Keymap(Keymap::kKeymapTypeGame, "blueforce-shortcuts", _("Blue force keymappings"));
+	Keymap *ringworldKeyMap = new Keymap(Keymap::kKeymapTypeGame, "ringworld-shortcuts", _("Ringworld keymappings"));
```

Comment:
The blueForceKeyMap and ringworldKeyMap do not appear to have assigned actions below, and they're not in the returned keymap arrray

Reply:
@OMGPizzaGuy fixed